src/tsolvers/egraph/EnodeStore.cc

```diff
@@ -150,9 +150,7 @@ vec<PTRefERefPair> EnodeStore::constructTerm(PTRef tr) {
 }
 
 bool EnodeStore::needsRecursiveDefinition(PTRef tr) const {
-    bool recdef = true;
-    for (auto ch : logic.getPterm(tr)) {
-        recdef &= needsEnode(ch);
-    }
-    return recdef;
+    Pterm const & t = logic.getPterm(tr);
+    opensmt::span const sp(t.begin(), t.size());
+    return std::all_of(sp.begin(), sp.end(), [&](PTRef ch) { return needsEnode(ch); });
```

Comment:
```suggestion
    return std::all_of(sp.begin(), sp.end(), [this](PTRef ch) { return needsEnode(ch); });
```
I like restricting the lambda's capture, if it is not too wordy. But I am also OK with your version.

Reply:
For me it's the same.  We can go with your suggestion.  I think the three extra characters will not cause major problems with modern disk capacity.